app/model/commands/UpdateAtomCommand.scala

```diff
@@ -69,6 +75,20 @@ case class UpdateAtomCommand(id: String, atom: MediaAtom, override val stores: D
     )
   }
 
+  private def processPlutoData(oldAtom: MediaAtom, newAtom: MediaAtom) = {
+    (oldAtom.plutoData.flatMap(_.projectId), newAtom.plutoData.flatMap(_.projectId)) match {
+      case (Some(oldProject), Some(newProject)) if oldProject != newProject => notifyPluto(newAtom)
+      case (None, Some(_)) => notifyPluto(newAtom)
+      case (_, _) => None
```

Comment:
If we don't do anything in the case of missing project ids here, are we still saving projects without pluto project ids to be looked at later?

Reply:
This serves to send a message to Pluto if the project property of an Atom has changed.

> are we still saving projects without pluto project ids to be looked at later?

Yes, however we do this on video upload as we talk to Pluto to tell it to ingest the video. During upload:

```
if !projectId set:
  [write to table](https://github.com/guardian/media-atom-maker/blob/master/common/src/main/scala/com/gu/media/upload/PlutoUploadActions.scala#L18-L19)
  send email to user
else:
  [send message](https://github.com/guardian/media-atom-maker/blob/master/common/src/main/scala/com/gu/media/upload/PlutoUploadActions.scala#L17) to pluto to ingest newly uploaded video
```

Doing so here would be a little noisy as its the UpdateCommand, so an edit to the Atom title, description, etc would mean also writing to that table...

`processPlutoData ` isn't the clearest naming (suggestions welcomed) and maybe something like this would be clearer:

```scala
(oldAtom.plutoData.flatMap(_.projectId), newAtom.plutoData.flatMap(_.projectId)) match {
  case (Some(oldProject), Some(newProject)) if oldProject == newProject => None
  case (_, Some(_)) => notifyPluto(newAtom)
  case (_, _) => None
}
```

Reply:
Ok, that makes sense! I guess the function could be called something like `processChangesToPlutoData` but not sure it makes much of a difference.